src/sys/unix.rs

```diff
@@ -867,6 +867,72 @@ impl crate::Socket {
         unsafe { setsockopt::<c_int>(self.inner, libc::SOL_SOCKET, libc::SO_MARK, mark as c_int) }
     }
 
+    /// Gets the value for the `SO_BINDTODEVICE` option on this socket.
+    ///
+    /// This value gets the socket binded device's interface name.
+    ///
+    /// This function is only available on Linux.
+    #[cfg(all(feature = "all", target_os = "linux"))]
+    pub fn device(&self) -> io::Result<Option<String>> {
+        unsafe {
+            let mut opt_len = libc::IFNAMSIZ as libc::socklen_t;
+            let mut payload: MaybeUninit<[u8; libc::IFNAMSIZ]> = MaybeUninit::uninit();
+            syscall!(getsockopt(
+                self.inner,
+                libc::SOL_SOCKET,
+                libc::SO_BINDTODEVICE,
+                payload.as_mut_ptr().cast(),
+                &mut opt_len,
+            ))?;
+            match opt_len {
+                0 => Ok(None),
+                _ => {
+                    let end: usize = opt_len as usize - 1;
+                    match String::from_utf8(payload.assume_init()[..end].to_vec()) {
```

Comment:
This is undefined behaviour. With `assume_init` your saying that `payload` is fully initialised, but only `opt_len` bytes are.

Reply:
Thank you for your advice. I haven't been proficient in using `slice` to interact...